pkg/windows/windows.go

```diff
@@ -335,6 +344,203 @@ func (vm *windows) Configure() error {
 	return vm.runBootstrapper()
 }
 
+// getClusterInfrastructureStatus returns a reference to the current InfrastructureStatus
+func getClusterInfrastructureStatus() (*config.InfrastructureStatus, error) {
+	cfg, err := crclientcfg.GetConfig()
+	if err != nil {
+		return nil, err
+	}
+	cs, err := clientset.NewForConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+	infra, err := cs.ConfigV1().Infrastructures().Get(context.TODO(), "cluster", meta.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return &infra.Status, nil
+}
+
+// clearDnsClientCache flushes the DNS client cache
+func (vm *windows) clearDnsClientCache() error {
+	vm.log.Info("Clearing DNS client cache", "vm", vm.hostName)
+	// flush DNS cache
+	if _, err := vm.Run("Clear-DnsClientCache", true); err != nil {
+		return errors.Wrap(err, "error clearing DNS client cache")
+	}
+	// return no error
+	return nil
+}
+
+// parseHostname returns a valid hostname given the full endpoint URL
+//
+// For example, given https://api.cluster.openshift.com:6443 the
+// value api.cluster.openshift.com is returned
+func parseHostname(urlString string) (string, error) {
+	if urlString == "" {
+		return "", errors.New("urlString cannot be empty")
+	}
+	u, err := url.Parse(urlString)
+	if err != nil {
+		return "", errors.Wrapf(err, "unable to parse URL: %s", urlString)
+	}
+	if u.Hostname() == "" {
+		return "", errors.Errorf("cannot to parse URL: %s", urlString)
+	}
+	// return hostname
+	return u.Hostname(), nil
+}
+
+// resolveIpAddresses returns a list of strings representing the IP addresses
+// associated with the given hostname. If noHostsFile==true the entries in the
+// Windows hosts file are skipped when resolving the DNS query
+func (vm *windows) resolveIpAddresses(hostname string, noHostsFile bool) ([]string, error) {
```

Comment:
the `noHostsFile` param can be removed, depending on [this](https://github.com/openshift/windows-machine-config-operator/pull/519/files#diff-4d8fe3bb66a872f0e4a277f820d60d2efa722f09aa89516be74b98c06b57f3bdR502)